### PR TITLE
feat(sequence): add more tests

### PR DESCRIPTION
**Pull request by connorskees**
Adds some more tests to sequence. Specifically some interval-related tests from another PR that got killed, and skipped tests related to grouped batching and reconnect, which currently has a number of crashes.

### CODE REVIEW

File: packages/dds/merge-tree/api-report/merge-tree.api.md [hunk 1]
```diff
@@ -249,9 +249,15 @@ export interface Dictionary<TKey, TData> {
     remove(key: TKey): void;
 }
 
+// @internal (undocumented)
+export function disableStrictPartialLengthChecks(): void;
```
Comment (Abe27342) [inline packages/dds/merge-tree/api-report/merge-tree.api.md:253]: can we export these from the test index only?

Comment (connorskees) [reply packages/dds/merge-tree/api-report/merge-tree.api.md:253]: Oh, can other packages now reach into `merge-tree/test`?

Comment (Abe27342) [reply packages/dds/merge-tree/api-report/merge-tree.api.md:253]: Yeah, see package.json exports:

image

so `import { disableStrictPartialLengthChecks } from "@fluidframework/merge-tree/dist/test"` should work.

we do this a few places in sequence and I think e2e tests to reuse some test utilities

Comment (Abe27342) [reply packages/dds/merge-tree/api-report/merge-tree.api.md:253]: might be nice to just expose this as a function which invokes both mocha hooks then, that should simplify some test code and help make sure we don't accidentally leave it on for a test suite.

Comment (connorskees) [reply packages/dds/merge-tree/api-report/merge-tree.api.md:253]: Oh, I didn't know you could programmatically invoke mocha hooks. If you call beforeEach twice, will both run or will one override the other?

File: packages/dds/sequence/src/test/fuzz/sharedString.fuzz.spec.ts [hunk 2]
```diff
@@ -103,3 +103,28 @@ describe("SharedString fuzz testing with rebased batches", () => {
 		// replay: 0,
 	});
 });
+
+// todo: potentially related to AB#7050
```
Comment (Abe27342) [inline packages/dds/sequence/src/test/fuzz/sharedString.fuzz.spec.ts:107]: Could you mention briefly that a few seeds are minimized and written out in intervalRebasing.spec.ts? Also probably good to write something similar near those tests

File: packages/dds/sequence/src/test/intervalRebasing.spec.ts [hunk 3]
```diff
@@ -480,4 +524,133 @@ describe("interval rebasing", () => {
...
+
+	it("delete and insert text into range containing interval while disconnected", async () => {
+		// 012
+		// (0)-x-12
+		const intervals = clients[0].sharedString.getIntervalCollection("comments");
+		clients[0].sharedString.insertText(0, "012");
+		intervals.add({ start: 0, end: 2, props: { intervalId: "0" } });
+		assertSequenceIntervals(clients[0].sharedString, intervals, [{ start: 0, end: 2 }]);
+
+		clients[0].containerRuntime.connected = false;
```
Comment (Abe27342) [inline packages/dds/sequence/src/test/intervalRebasing.spec.ts:644]: in general with the way the mocks work, this should function equivalently to if you move it to the start of the test (disconnecting a client will cancel all their in-flight ops in the mock implementation), which I generally find to be less confusing to reason about: as written it looks like there might be a difference between the interval addition and the insert of "x", but there isn't really.

File: packages/dds/merge-tree/api-report/merge-tree.api.md (hunk 1 above)
Comment (Abe27342) [reply packages/dds/merge-tree/api-report/merge-tree.api.md:253]: They'll both be run. Mocha runs all hooks for each type in the order they're defined, so outermost suite aka global hooks -> innermost. I get the impression the implementation just adds it to a list on the test run state object. Never bothered to look though. See [here](https://mochajs.org/#hooks).

Comment (Abe27342) [reply packages/dds/merge-tree/api-report/merge-tree.api.md:253]: You can do some neat stuff with that structure btw, see e.g. https://github.com/microsoft/FluidFramework/pull/18919/files

**Review by Abe27342 — APPROVED** [review]

**Pull request merged**